src/ParaTest/Logging/JUnit/Writer.php

```diff
@@ -147,7 +147,7 @@ protected function appendCase($suiteNode, TestCase $case)
     protected function appendDefects($caseNode, $defects, $type)
     {
         foreach($defects as $defect) {
-            $defectNode = $this->document->createElement($type, $defect['text'] . "\n");
+            $defectNode = $this->document->createElement($type, htmlentities($defect['text']) . "\n");
```

Comment:
While this works for the case when there's only `&` in the string, this breaks other cases when there are other HTML entities.

For example: if the text has `—`, then this line change converts it to `&mdash;` which the breaks the junit.xml parsing.

Reply:
A better fix is to use `htmlspecialchars` which only escapes the necessary characters.

Reply:
See #296